Automattic/woocommerce-payments
# Prevent processing the wrong kind of webhook for a gateway

Fixes #5266 

#### Changes proposed in this Pull Request

- Before processing a webhook event, we verify that the event's mode (defined by the `livemode` flag in the event's payload) is identical to the gateway's mode. An exception is thrown if they don't match.

#### Testing instructions

As described in the issue:

* We need a live site
* Add a failed order to the live site
* Clone the live site to create a staging site
* Pay for the failed order on the staging site
* The order on the live site should still be marked as failed

The live and clone site should have WCPay dev installed so that we can redirect the API to the server (exposed through ngrok) using that branch

-------------------


**Post merge**
